sa/sa.go

```diff
@@ -402,6 +414,46 @@ func (ssa *SQLStorageAuthority) countCertificatesByName(domain string, earliest,
 	return len(serialMap), nil
 }
 
+// countCertificatesByExactNames returns, for a single domain, the count of
+// certificates issued in the given time range for that domain. In contrast to
+// the countCertificatesByNames function subdomains are NOT considered.
+//
+// The highest count this function can return is 10,000. If there are more
+// certificates than that matching one of the provided domain names, it will return
+// TooManyCertificatesError.
+func (ssa *SQLStorageAuthority) countCertificatesByExactName(domain string, earliest, latest time.Time) (int, error) {
```

Comment:
We can collapse this function with countCertificatesByExactName if we parameterize the `SELECT ...` statement. For instance, we could offer two constants: exactSelect and subdomainSelect, and pass those as arguments to a single function.

Also, right now countCertificatesByName has `WHERE reversedName = :reversedDomain OR reversedName LIKE CONCAT(:reversedDomain, ".%"))`. However, maybe we should remove that first clause so that these counts are non-overlapping? In other words, we have decided we want to treat "exact public suffixes" as being in their own limiting bucket. So we shouldn't count them against the limiting bucket for subdomain matches.